pkg/restore/restore.go

```diff
@@ -149,23 +155,80 @@ func NewKubernetesRestorer(
 	resourcePriorities []string,
 	backupClient arkv1client.BackupsGetter,
 	namespaceClient corev1.NamespaceInterface,
+	namespaceInformer corev1informers.NamespaceInformer,
+	pvClient corev1.PersistentVolumeInterface,
+	pvInformer corev1informers.PersistentVolumeInformer,
 	resticRestorerFactory restic.RestorerFactory,
 	resticTimeout time.Duration,
+	namespaceTimeout time.Duration,
 	logger logrus.FieldLogger,
 ) (Restorer, error) {
-	return &kubernetesRestorer{
+	kr := &kubernetesRestorer{
 		discoveryHelper:       discoveryHelper,
 		dynamicFactory:        dynamicFactory,
 		blockStore:            blockStore,
 		backupClient:          backupClient,
 		namespaceClient:       namespaceClient,
+		pvClient:              pvClient,
 		resticRestorerFactory: resticRestorerFactory,
 		resticTimeout:         resticTimeout,
+		namespaceTimeout:      namespaceTimeout,
+		nsWaiter:              &namespaceWaiter{chans: make(map[string]chan *v1.Namespace)},
+		pvWaiter:              &pvWaiter{chans: make(map[string]chan *v1.PersistentVolume)},
 		resourcePriorities:    resourcePriorities,
 		logger:                logger,
 
 		fileSystem: filesystem.NewFileSystem(),
-	}, nil
+	}
+
+	namespaceInformer.Informer().AddEventHandler(
```

Comment:
Upon some reflection, I'd like to have test coverage for these two event handlers, but the two solutions I came up with to do so seemed kind of heavy weight considering what they do.

* Mock the namespace and PV informers, such that the tests will add/remove/change objects so the handlers will be triggered. This is more like what our code currently does, but is likely much more scaffolding.
* Instead of doing anonymous functions with closures over `kr`, make a factory function that will wrap `kr` and return a handler function. This means all we'd need to create to get handlers to test would be a `KubernetesRestorer`, and the functions can be called directly with dummy objects. This is a little further removed from the actual code, though.